### PR TITLE
[FLINK-8530] [flip6] Enable detached job mode submission to session cluster

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

This commit makes the RestClusterClient aware whether the user wishes to submit
a job in detached or non-detached mode. If it is detached, then the RestClusterClient
won't poll for the execution result.

This PR is based on #5428.

## Verifying this change

- Added `RestClusterClientTest#testDetachedJobSubmission`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java [hunk 1]
```diff
@@ -456,7 +483,7 @@ public void testTriggerSavepoint() throws Exception {
 					return CompletableFuture.completedFuture(
 						new SavepointTriggerResponseBody(testSavepointTriggerId));
 				} else {
-					// return new random savepoint trigger id so that test can fail
+					// return new random savepoint trigger jobId so that test can fail
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: no

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java [hunk 2]
```diff
@@ -493,7 +520,7 @@ public void testTriggerSavepoint() throws Exception {
 				} else {
 					return FutureUtils.completedExceptionally(
 						new RestHandlerException(
-							"Unexpected savepoint trigger id: " + savepointTriggerId,
+							"Unexpected savepoint trigger jobId: " + savepointTriggerId,
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: something went wrong here

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java [hunk 3]
```diff
@@ -509,7 +536,7 @@ public void testListJobs() throws Exception {
 				Iterator<JobStatusMessage> jobDetailsIterator = jobDetails.iterator();
 				JobStatusMessage job1 = jobDetailsIterator.next();
 				JobStatusMessage job2 = jobDetailsIterator.next();
-				Assert.assertNotEquals("The job statues should not be equal.", job1.getJobState(), job2.getJobState());
+				Assert.assertNotEquals("The jobGraph statues should not be equal.", job1.getJobState(), job2.getJobState());
```
Comment (GJL) [inline flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: previous message was fine ... also *status*

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: True, will fix it.

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java (hunk 2 above)
Comment (tillrohrmann) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: Will fix it.

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java (hunk 3 above)
Comment (tillrohrmann) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: Will fix it.

**Comment by tillrohrmann** [comment]
Thanks for the review @GJL. I've addressed your feedback and rebased onto the latest soon to be master.

**Pull request closed without merging**